client/blocks/credit-card/use-payment-intents.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').EmitResponseProps} EmitResponseProps
+ * @typedef {import('../stripe-utils/type-defs').Stripe} Stripe
+ */
+
+/**
+ * Opens the modal for PaymentIntent authorizations.
+ *
+ * @param {Object}           params                Params object.
+ * @param {Stripe}           params.stripe         The stripe object.
+ * @param {Object}           params.paymentDetails The payment details from the
+ *                                                 server after checkout processing.
+ * @param {string}           params.errorContext   Context where errors will be added.
+ * @param {string}           params.errorType      Type of error responses.
+ * @param {string}           params.successType    Type of success responses.
+ */
+const openIntentModal = ( {
+	stripe,
+	paymentDetails,
+	errorContext,
+	errorType,
+	successType,
+} ) => {
+	const checkoutResponse = { type: successType };
+	if (
+		! paymentDetails.setup_intent &&
+		! paymentDetails.payment_intent_secret
+	) {
+		return checkoutResponse;
+	}
+	const isSetupIntent = !! paymentDetails.setupIntent;
```

Comment:
I would be more comfortable with a `undefined !== paymentDetails.setupIntent` or `null !== paymentDetails.setupIntent`. The `!!` is a bit confusing at first, at JS's type coercion doesn't always follow what common sense might dictate 🙂

Reply:
You can make that change if you want. There already is a truthy check before this line, so this is explicitly casting to a boolean for future usage of `isSetupIntent`.

Reply:
Let's keep this as is for now. I'm marking this as resolved.